ansible/playbooks/roles/ocp-config/tasks/main.yaml

```diff
@@ -1,5 +1,54 @@
 ---
 # tasks file for ocp4 config
+- name: Extract OCP4 tools
+  when: release_image_override != ""
+  block:
+    - name: Set tools directory path
+      set_fact:
+        tools_dir: "~/ocp4_tools"
+
+    - name: Create tools directory
+      file:
+        path: "{{ tools_dir }}"
+        state: "{{ item }}"
+        mode: '0755'
+      with_items:
+      - absent
+      - directory
+
+    - name: Extract OCP4 tools from release image
+      shell: |
+        echo '{{ install_config.pull_secret }}' > pull_secret
+        oc adm release extract --tools {{ release_image_override }} --registry-config='pull_secret'
+      args:
+        chdir: "{{ tools_dir }}"
+
+    - name: Remove exsisting binaries
+      file:
+        path: "/usr/local/bin/{{ item }}"
+        state: absent
+        mode: '0755'
+      with_items:
+      - oc
+      - openshift-install
```

Comment:
Can't we just overwrite the existing ones with new binaries and avoid creating another set of directories ?